src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -0,0 +1,389 @@
+
+#ifdef _WIN32
+/*
+ * Check the architectural compatability(32bit vs 64bit) between the app and
+ * lib.
+ */
+static bool check_compatibility(HANDLE parent, HANDLE child)
+{
+  static launcher& app = launcher::get_instance();
+  BOOL is_parent_wow64 = FALSE, is_child_wow64 = FALSE;
+
+  IsWow64Process(parent, &is_parent_wow64);
+  IsWow64Process(child, &is_child_wow64);
+
+  if (is_parent_wow64 != is_child_wow64)
+  {
+    log_e(app.m_name, "is", (is_parent_wow64 ? w64 : w32),
+          "-bit but target application is ", (is_child_wow64 ? w64 : w32),
+          "-bit");
+    return false;
+  }
+
+  return true;
+}
+
+/*
+ * Function to get the current precise system time as pair for string and
+ * formated string
+ */
+std::pair<std::string, std::string>
+get_current_time_as_string()
+{
+  FILETIME ft;
+  GetSystemTimePreciseAsFileTime(&ft);
+
+  ULARGE_INTEGER uli;
+  uli.LowPart = ft.dwLowDateTime;
+  uli.HighPart = ft.dwHighDateTime;
+
+  // Convert the time to a string
+  std::ostringstream oss;
+  oss << uli.QuadPart;
+
+  // Time formatting and trace directory printing
+  FILETIME local_ft;
+  FileTimeToLocalFileTime(&ft, &local_ft);
+
+  SYSTEMTIME st;
+  FileTimeToSystemTime(&local_ft, &st);
+
+  std::ostringstream ss;
+  ss << std::setfill('0') << std::setw(4) << st.wYear << "-" << std::setw(2)
+     << st.wMonth << "-" << std::setw(2) << st.wDay << "_" << std::setw(2)
+     << st.wHour << "-" << std::setw(2) << st.wMinute << "-" << std::setw(2)
+     << st.wSecond;
+
+  std::string formatted_time = ss.str();
+
+  return {oss.str(), formatted_time};
+}
+
+int create_child_proc_as_suspended(launcher& app)
+{
+  // Create child process in suspended state:
+  log_d("Creating child process with command line: ", app.m_cmdline);
+
+  // Initialize si and pi
+  ZeroMemory(&app.m_si, sizeof(STARTUPINFOA));
+  app.m_si.cb = sizeof(STARTUPINFOA);
+  ZeroMemory(&app.m_pi, sizeof(PROCESS_INFORMATION));
+
+  if (CreateProcessA(
+      NULL,                         // No module name (use command line)
+      (LPSTR)app.m_cmdline.c_str(), // Command line
+      NULL,                         // Process handle not inheritable
+      NULL,                         // Thread handle not inheritable
+      FALSE,                        // Set handle inheritance to FALSE
+      CREATE_SUSPENDED,             // Process created in a suspended state
+      NULL,                         // Use parent's environment block
+      NULL,                         // Use parent's starting directory
+      &app.m_si,                    // Pointer to STARTUPINFO structure
+      &app.m_pi) == FALSE)          // Pointer to PROCESS_INFORMATION structure
+  {
+    log_f("Child process creation failed");
+    return -1;
+  }
+
+  log_d("Child process created. ");
+
+  return 0;
+
+}
+
+int resume_child_proc_and_wait_for_completion(launcher& app)
+{
+  DWORD retval = 0;
+  if (ResumeThread(app.m_pi.hThread) < 0) {
+    log_e("Failed to resume thread");
+    return -1;
+  }
```

Comment:
The bracing is golden for me :-), but you have to be consistent throughout all the files.